src/main/java/org/folio/circulation/domain/UpdateItem.java

```diff
@@ -44,7 +52,13 @@ private Item changeItemOnCheckIn(
 
     if (requestQueue.hasOutstandingFulfillableRequests()) {
       Request request = requestQueue.getHighestPriorityFulfillableRequest();
-      UUID pickUpServicePointId = UUID.fromString(request.getPickupServicePointId());
+
+      String pickupServicePointIdString = request.getPickupServicePointId();
+      if (pickupServicePointIdString == null) {
+        return null;
```

Comment:
I think eventually it would be better to convert this method into responding with a result, and raise the error here instead.

What do you think?

Reply:
Yup. As I mentioned in @mtraneis' resolved comment, I don't really like using `null` here. Should I just fix it right or do we want to get this in now so that we don't get the ugly NPE and come up with a better approach later?

Reply:
I've pushed a branch demonstrating the alternative - https://github.com/folio-org/mod-circulation/tree/CIRC-227-alternative

Reply:
https://github.com/folio-org/mod-circulation/blob/a6e4c36e2206647a43c894b2dd0808b3d6d1d4b2/src/main/java/org/folio/circulation/domain/UpdateItem.java#L51

Reply:
https://github.com/folio-org/mod-circulation/blob/a6e4c36e2206647a43c894b2dd0808b3d6d1d4b2/src/main/java/org/folio/circulation/domain/UpdateItem.java#L51-L56

Reply:
Ah. Of course! That's what I get for rushing through this last night.

Reply:
OK. I've made these changes. The code looks much better now... well, except for the long error message. :)